custom_components/tahoma/binary_sensor.py

```diff
@@ -61,10 +60,10 @@
 async def async_setup_entry(hass, entry, async_add_entities):
     """Set up the TaHoma sensors from a config entry."""
     data = hass.data[DOMAIN][entry.entry_id]
-    controller = data.get("controller")
+    coordinator = data.get("coordinator")
 
     entities = [
-        TahomaBinarySensor(device, controller)
+        TahomaBinarySensor(device.url, coordinator)
```

Comment:
device.url?

Reply:
You are right. I think I will create a PR request to rename device.deviceurl to device.url, make more sense and we don't repeat ourself.